### DISCUSSION TITLE
Possibility to reference remote data sources?

### DISCUSSION THREAD

**Discussion by manuschillerdev** [open-policy-agent/community#405] [OPA and Rego] [answered]
I am wondering if it would be possible to reference remote data sources in rego rules - and if this would be desirable at all.

Example:

```rego
package example
import future.keywords 

allow := true {                         
    count(violation) == 0         
}
violation[server.id] {                  
    # hypothetical keyword "remote"
    some server in remote "http://some-serivce.my-ns.svc.cluster.local/data-endpoint/public_servers"  # returns a JSON-Reponse with an array
    "http" in server.protocols          # it contains the insecure "http" protocol.
}
```

this obviously would have some things to consider like
* stability/Reproducability of policies
* latency?
* probably a whole bunch of others

Is this a thing in Rego?

**Comment by anderseknert** [comment, marked as answer]
Might be I don’t understand, but would the http.send builtin function be what you’re looking for?

https://www.openpolicyagent.org/docs/latest/policy-reference/#http

**Reply by manuschillerdev** [reply]
Yes! I totally missed that - thanks a lot!
